src/hotspot/share/opto/loopopts.cpp

```diff
@@ -341,24 +341,9 @@ void PhaseIdealLoop::dominated_by(IfProjNode* prevdom, IfNode* iff, bool flip, b
   // dominating projection.
   Node* dp = iff->proj_out_or_null(pop == Op_IfTrue);
 
-  // Loop predicates may have depending checks which should not
-  // be skipped. For example, range check predicate has two checks
-  // for lower and upper bounds.
   if (dp == nullptr)
     return;
 
-  ProjNode* dp_proj  = dp->as_Proj();
-  ProjNode* unc_proj = iff->proj_out(1 - dp_proj->_con)->as_Proj();
-  if (exclude_loop_predicate &&
-      (unc_proj->is_uncommon_trap_proj(Deoptimization::Reason_predicate) != nullptr ||
-       unc_proj->is_uncommon_trap_proj(Deoptimization::Reason_profile_predicate) != nullptr ||
-       unc_proj->is_uncommon_trap_proj(Deoptimization::Reason_range_check) != nullptr)) {
-    // If this is a range check (IfNode::is_range_check), do not
-    // reorder because Compile::allow_range_check_smearing might have
-    // changed the check.
-    return; // Let IGVN transformation change control dependence.
-  }
-
```

Comment:
Why it ok to remove this bailout?

Reply:
It's:
"IfNode::dominated_by() and PhaseIdealLoop::dominated_by() have
logic to prevent this: nodes that are control dependent on a range
check or predicate are not allowed to float."
that I mentioned in the fix description. It's the way array access nodes are currently prevented from floating above the range checks they depend on. It's flawed, replaced by pinning of the array access nodes in the patch. So this logic is no longer useful.